clippy_utils/src/qualify_min_const_fn.rs

```diff
@@ -245,33 +253,33 @@ fn check_operand<'tcx>(tcx: TyCtxt<'tcx>, operand: &Operand<'tcx>, span: Span, b
                 ));
             }
 
-            check_place(tcx, *place, span, body)
+            check_place(tcx, *place, span, body, msrv)
         },
-        Operand::Copy(place) => check_place(tcx, *place, span, body),
+        Operand::Copy(place) => check_place(tcx, *place, span, body, msrv),
         Operand::Constant(c) => match c.check_static_ptr(tcx) {
             Some(_) => Err((span, "cannot access `static` items in const fn".into())),
             None => Ok(()),
         },
     }
 }
 
-fn check_place<'tcx>(tcx: TyCtxt<'tcx>, place: Place<'tcx>, span: Span, body: &Body<'tcx>) -> McfResult {
+fn check_place<'tcx>(tcx: TyCtxt<'tcx>, place: Place<'tcx>, span: Span, body: &Body<'tcx>, msrv: &Msrv) -> McfResult {
     for (base, elem) in place.as_ref().iter_projections() {
         match elem {
             ProjectionElem::Field(..) => {
-                let base_ty = base.ty(body, tcx).ty;
-                if let Some(def) = base_ty.ty_adt_def() {
-                    // No union field accesses in `const fn`
-                    if def.is_union() {
-                        return Err((span, "accessing union fields is unstable".into()));
-                    }
+                if base.ty(body, tcx).ty.is_union() && !msrv.meets(msrvs::CONST_FN_UNION) {
+                    return Err((span, "accessing union fields is unstable".into()));
+                }
+            },
+            ProjectionElem::Deref => {
+                if base.ty(body, tcx).ty.is_unsafe_ptr() && !msrv.meets(msrvs::CONST_RAW_PTR_DEREF) {
```

Comment:
Dereferencing `*mut T` is still unstable today (https://github.com/rust-lang/rust/issues/57349) so I think this should only allow `RawPtr(Mutability::Not)` specifically and always error for `RawPtr(Mutability::Mut)`
(sorry if I suggested `is_unsafe_ptr` in the first place 😅)

Reply:
Oh, right, I forgot about the mut pointers as well 😄